tests/test_hyperparameter_search.py

```diff
@@ -0,0 +1,223 @@
+    HuggingFaceClassificationPipelineMetadata,
+    HuggingFacePairClassificationPipelineMetadata,
+    HuggingFaceSequenceLabelingPipelineMetadata,
+)
+from embeddings.utils.utils import PrimitiveTypes
+
+
+@pytest.fixture
+def output_path() -> TemporaryDirectory[str]:
+    return TemporaryDirectory()
+
+
+@pytest.fixture
+def text_classification_dataset_kwargs() -> Dict[str, PrimitiveTypes]:
+    return {
+        "dataset_name": "clarin-pl/polemo2-official",
+        "input_column_name": "text",
+        "target_column_name": "target",
+        "load_dataset_kwargs": None,
+    }
+
+
+@pytest.fixture
+def text_pair_classification_dataset_kwargs() -> Dict[str, Union[PrimitiveTypes, Tuple[str, str]]]:
+    return {
+        "dataset_name": "clarin-pl/cst-wikinews",
+        "input_columns_names_pair": ("sentence_1", "sentence_2"),
+        "target_column_name": "label",
+        "load_dataset_kwargs": None,
+    }
+
+
+@pytest.fixture
+def sequence_labeling_dataset_kwargs() -> Dict[str, PrimitiveTypes]:
+    return {
+        "dataset_name": "clarin-pl/kpwr-ner",
+        "input_column_name": "tokens",
+        "target_column_name": "ner",
+        "load_dataset_kwargs": None,
+    }
+
+
+@pytest.fixture
+def pipeline_kwargs(output_path: TemporaryDirectory[str]) -> Dict[str, PrimitiveTypes]:
+    return {
+        "output_path": output_path.name,
+        "embedding_name": "clarin-pl/roberta-polish-kgr10",
+        "document_embedding_cls": "FlairDocumentPoolEmbedding",
+        "load_model_kwargs": None,
+        "task_model_kwargs": None,
+        "task_train_kwargs": None,
+    }
+
+
+@pytest.fixture
+def sequence_labeling_pipeline_kwargs() -> Dict[str, PrimitiveTypes]:
+    return {"evaluation_mode": "conll", "tagging_scheme": None, "hidden_size": 128}
+
+
+@pytest.fixture
+def hf_classification_pipeline_metadata(
+    pipeline_kwargs: Dict[str, str],
+    text_classification_dataset_kwargs: Dict[str, PrimitiveTypes],
+) -> Dict[str, Any]:
+    return {
+        **pipeline_kwargs,
+        **text_classification_dataset_kwargs,
+    }
+
+
+@pytest.fixture
+def hf_pair_classification_pipeline_metadata(
+    pipeline_kwargs: Dict[str, str],
+    text_pair_classification_dataset_kwargs: Dict[str, PrimitiveTypes],
+) -> Dict[str, Any]:
+    return {
+        **pipeline_kwargs,
+        **text_pair_classification_dataset_kwargs,
+    }
+
+
+@pytest.fixture
+def hf_sequence_labeling_pipeline_metadata(
+    pipeline_kwargs: Dict[str, str],
+    sequence_labeling_pipeline_kwargs: Dict[str, PrimitiveTypes],
+    sequence_labeling_dataset_kwargs: Dict[str, PrimitiveTypes],
+) -> Dict[str, Any]:
+    return {
+        **pipeline_kwargs,
+        **sequence_labeling_pipeline_kwargs,
+        **sequence_labeling_dataset_kwargs,
+    }
+
+
+def test_hf_classification_pipeline_metadata(
+    hf_classification_pipeline_metadata: Dict[str, Any]
+) -> None:
+    metadata = create_model_from_typeddict(HuggingFaceClassificationPipelineMetadata)(  # type: ignore
```

Comment:
why type ignore? maybe add justification in comment

Reply:
Added